viewing_party/party.py

```diff
@@ -1,23 +1,189 @@
+        return 0
+    for movie in janes_data["watched"]:
+        sum += movie["rating"]
+    return sum/(len(janes_data["watched"]))
+    pass
+
+#DONE
+def get_most_watched_genre(data):
+    genreDict = {}
+    for movie in data["watched"]:
+        genre = movie["genre"]
+        if genre not in genreDict:
+            genreDict[genre] = 1
+        elif genre in genreDict:
+            genreDict[genre] += 1 
+    maxValue = 0
+    for value in genreDict.values():
+        if value > maxValue:
+            maxValue = value
+    for key,value in genreDict.items():
+        if value == maxValue:
+            return key   
+    pass
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+import copy 
+
+#DONE
+def get_unique_watched(data):
+    #amandaUniqueMovieList = []
+
+    #can't do it this way because it changes the original list
+    #anotherList = data["watched"]
+
+    #list of movies amanda has seen
+    #print(data["watched"])
+
+    #this is a list of the first friend's watched movies
+    #print(data["friends"][0]["watched"])
+    
+    #this is a list of the second friend's watched movies
+    #print(data["friends"][1]["watched"])
+    
+    #newList = []
+
+    origCopy = data["watched"]
+    listCopy = copy.deepcopy(origCopy)
+    for j in range(len(data["friends"])):
+        for i in range(len(data["friends"][j]["watched"])):
+            if data["friends"][j]["watched"][i] in listCopy:
+                listCopy.remove(data["friends"][j]["watched"][i])
+    return listCopy
+            #this prints all the movies in the friends watch list from every dictionary in the list
+            #print(data["friends"][j]["watched"][i]["title"])
+
+            #this prints all the dictionaries of movies all friends have watched 
+            #print(data["friends"][j]["watched"][i])
+
+            #this is giving me the wrong list, it's a list of movies everyone has seen 
+            #if data["friends"][j]["watched"][i] in data["watched"]:
+                #newList.append(data["friends"][j]["watched"][i])
+    #print(newList)
+            
+            #can't do it this way because it changes the original list
+            #if data["friends"][j]["watched"][i] in anotherList:
+                #anotherList.remove(data["friends"][j]["watched"][i])
+    #print(anotherList)
+
+
+    #this was wrong
+    #for movie in data["watched"]:
+        #if movie["title"] not in data["friends"]:
+            #amandaUniqueMovieList.append(movie)
+    #print(amandaUniqueMovieList)
+
+    pass
+
+#DONE!
+def get_friends_unique_watched(data):
+    newList = []
+    for j in range(len(data["friends"])):
+        for i in range(len(data["friends"][j]["watched"])):
+            if data["friends"][j]["watched"][i] not in data["watched"]:
+                if data["friends"][j]["watched"][i] not in newList:
+                    newList.append(data["friends"][j]["watched"][i])

```

Comment:
Because we want to append values to the `newList`, we can use a `for in` loop to iterate through the friends watched lists.